utils/patternGenerator.ts

```diff
@@ -0,0 +1,141 @@
+import parseYaml from "../utils/parseYaml";
+
+function getType($ref) {
+  const $refDirName = $ref.split("/");
+  return $refDirName[$refDirName.length - 1];
+}
+
+function getRpcDefinition(rpcDefinitionPath, rpcName) {
+  return parseYaml(rpcDefinitionPath).find(item => item.name === rpcName);
+}
+
+function getSchema() {
+  const baseTypes = parseYaml("../execution-apis/src/schemas/base-types.yaml");
+  const block = parseYaml("../execution-apis/src/schemas/block.yaml");
+  const client = parseYaml("../execution-apis/src/schemas/client.yaml");
+  const filter = parseYaml("../execution-apis/src/schemas/filter.yaml");
+  const receipt = parseYaml("../execution-apis/src/schemas/receipt.yaml");
+  const state = parseYaml("../execution-apis/src/schemas/state.yaml");
+  const transaction = parseYaml("../execution-apis/src/schemas/transaction.yaml");
+  const withdrawal = parseYaml("../execution-apis/src/schemas/withdrawal.yaml");
+
+  return {
+    ...baseTypes,
+    ...block,
+    ...client,
+    ...filter,
+    ...receipt,
+    ...state,
+    ...transaction,
+    ...withdrawal,
+  }
+}
+
+function isMainType(type) {
+  return type.charAt(0) === type.charAt(0).toUpperCase();
+}
+
+function getPattern(type) {
+  const schema = getSchema();
+  const schemaType = schema[type];
+  if (!isMainType(type) && schemaType) return new RegExp(schemaType.pattern);
+
+  const { properties } = schemaType;
+  if (properties) return iterateObjectProperties({}, properties);
+
+  let combinedPattern = {};
+  let combinedProperties;
+  const { allOf, oneOf } = schemaType;
+  if (allOf || oneOf) {
+    combinedProperties = allOf || oneOf;
+    combinedProperties.forEach((item) => {
+      if (item.properties) combinedPattern = { ...iterateObjectProperties({}, item.properties) }
+    });
+    return combinedPattern;
+  }
+
+  return null;
+}
+
+function iterateObjectProperties(pattern, properties) {
```

Comment:
We are always passing in `{}` for the first parameter

Reply:
ah yeah, this can just be within the function itself and not be listed as a parameter. i'll make an update